### PR TITLE
Cumprod support

### PR DESCRIPTION
**Pull request by reddyn12**
Works with CPU=1, but getting 0's for gpu/cuda. Need to find where 0 is being added to multiply list, but Idk where the code is compiled

### CODE REVIEW

**Comment by reddyn12** [comment]
@geohot Is this fine?

**Comment by reddyn12** [comment]
@chenyuxyz am I supposed to add anything else to the pr?

**Comment by chenyuxyz** [comment]
have you read this https://github.com/tinygrad/tinygrad/?tab=readme-ov-file#contributing? it also contains too many copy-pasted lines

**Comment by reddyn12** [comment]
@chenyuxyz I added tests, but they fail when ran on testing sever. It works on my system. Could it be because the new tests are being run on a precompiled version of tinygrad? Getting error that ReduceOs.PROD isn't a key, but made sure linearizer and ops_cpu have it in their respective dicts.

**Comment by reddyn12** [comment]
Ok I added the tests and made cumulative reduce a function. Is there anything I'm missing @chenyuxyz ?

**Review by reddyn12 — COMMENTED** [review]
Ready for review

File: test/test_tensor.py [hunk 1]
```diff
@@ -314,6 +314,15 @@ def test_item_to_tensor_to_item(self):
       reshaped_item = Tensor([a]).reshape((1, 1, 1, 1, 1)).item()
       assert type(reshaped_item) == type(a), a
       np.testing.assert_allclose(reshaped_item, a), a
+  def test_cumsum(self):
```
Comment (chenyuxyz) [inline test/test_tensor.py:317]: check how cumsum was tested in test_ops.py, it needs to test backward too

**Comment by geohot** [comment]
Tests failing, seems stale. Can reopen if you still want it

**Pull request closed without merging**